backend/src/openarchiefbeheer/destruction/tests/test_serializers.py

```diff
@@ -478,6 +479,97 @@ def test_if_user_not_assigned_cannot_create_review(self):
             ),
         )
 
+    @override_settings(LANGUAGE_CODE="en")
+    def test_if_list_in_wrong_status_cannot_be_reviewed(self):
+        user = UserFactory.create(
+            username="reviewer1",
+            email="reviewer1@oab.nl",
+            role__can_review_destruction=True,
+            role__can_review_final_list=True,
+        )
+        destruction_list = DestructionListFactory.create(
+            assignee=user, status=ListStatus.changes_requested
+        )
+
+        data = {
+            "destruction_list": destruction_list.uuid,
+            "decision": ReviewDecisionChoices.accepted,
+        }
+        request = factory.get("/foo")
+        request.user = user
+
+        serializer = DestructionListReviewSerializer(
+            data=data, context={"request": request}
+        )
+
+        self.assertFalse(serializer.is_valid())
+        self.assertEqual(
+            serializer.errors["author"][0],
+            _(
+                "The status of this destruction list prevents you from creating a review at this stage."
+            ),
+        )
+
+    @override_settings(LANGUAGE_CODE="en")
+    def test_if_user_not_a_reviewer_cannot_create_review(self):
+        user = UserFactory.create(
+            username="reviewer1",
+            email="reviewer1@oab.nl",
+            role__can_review_destruction=False,
+        )
+        destruction_list = DestructionListFactory.create(
+            assignee=user, status=ListStatus.ready_to_review
+        )
+
+        data = {
+            "destruction_list": destruction_list.uuid,
+            "decision": ReviewDecisionChoices.accepted,
+        }
+        request = factory.get("/foo")
+        request.user = user
+
+        serializer = DestructionListReviewSerializer(
+            data=data, context={"request": request}
+        )
+
+        self.assertFalse(serializer.is_valid())
+        self.assertEqual(
+            serializer.errors["author"][0],
+            _(
+                "The status of this destruction list prevents you from creating a review at this stage."
```

Comment:
In this case it is not really the status that prevents the user from creating a review, but more their role :thinking: I think that we can make the error vaguer ("you cannot create a review") since if things go well in the UI these cases shouldn't happen. And if someone is maliciously trying to create a review when they can't, then they get a vague error message?

Reply:
I don't really like vague error messages, they should indicate the problem so I t can be fixed. The user's role typically doesn't change. The status of a list does, therefore it's more likely that the issues is with the status. This error may occur when a list has changed status while a browser window is still open. In such cases a clear error response should be returned.

Also: regardless of the situation: either a reviewer wants to review a final list, or the archivist wants to finalize a list ready for review. The status prevents the user from doing its thing.

Reply:
But in these test cases the list is in the right status to receive a review from a reviewer, but it is the wrong reviewer/person with wrong permissions to attempt reviewing. So I think that the problem is not "the status" :thinking: 

I am thinking about the case where a window has been left open: normally if a user is assigned to a list, only their action can make the list change status. However, if a user is not assigned to it, in the UI it is disabled. If the status changes under water, in the browser it remains disabled.

The only case I can think where this would be a problem is if the record manager reassigns the list because the original reviewer is not doing their task. But then in this case the problem is still not the status, but that the user is not assigned :thinking:

Reply:
```
Feature: System should be able to deal with outdated statusses when creating a review.

Scenario: Outdated status
Given record manager creates a list
And the list is assigned to reviewer1 and reviewer2 for review
And reviewer1 opens the list
And reviewr 1 approves the list
And reviewer2 opens the list
And record manager reassignes the list from reviewer2 to reviewer3
And reviewer3 opens the list
And reviewer3 approves the list
When reviewer2 approves the list 
Then the status should be "ready for destruction"
Then a message should appear with the text "The status of this destruction list prevents you from creating a review at this stage."
```

Reply:
Urgh, we should maybe create an issue about this :S I still feel that the error message is not ideal, but if you are convinced that it is best like this, I am ok with it